### PR TITLE
Jormungandr: change /places authentication

### PR DESCRIPTION
**Pull request by antoine-de**
change the way the authentication works for the new autocomplete (bragi)

the API is still restricted to user that can access the open data

And we give all the user's coverages to the underlying autocomplete service (bragi)

### CODE REVIEW

File: source/jormungandr/tests/bragi_autocomplete_tests.py [hunk 1]
```diff
@@ -416,7 +416,8 @@ def test_places_with_empty_coord(self):
 
     def test_global_place_uri(self):
         mock_requests = MockRequests({
-            'https://host_of_bragi/features/bob': (BRAGI_MOCK_RESPONSE, 200)
+            # the pt_dataset is added because there is no authentication so all the known pt_dataset are added
```
Comment (pbougue) [inline source/jormungandr/tests/bragi_autocomplete_tests.py]: minor: not super clear that the pt_dataset mentioned first is "all the know dataset"

File: source/jormungandr/jormungandr/authentication.py [hunk 2]
```diff
@@ -135,26 +135,6 @@ def has_access(region, api, abort, user):
...
-    Since there is only open data in the global /places by default, to access the global places the user
-    should be either a super user or have access to the open data
-    """
-    if current_app.config.get('PUBLIC', False):
-        #if jormungandr is on public mode we skip the authentification process
-        return True
-    if not user:
-        # a user is mandatory even for free region
-        return False
-    return user.is_super_user or user.have_access_to_free_instances
```
Comment (pbougue) [inline source/jormungandr/jormungandr/authentication.py:150]: Now sandbox token is accessing places... We don't want that, and we have to discuss a bit about that (probably just authorize if you have access to open data).
> ✔️ Done : discussed about it, back to old way of working for that

File: source/jormungandr/jormungandr/authentication.py [hunk 3]
```diff
@@ -39,7 +39,7 @@
 import datetime
 import base64
 from navitiacommon.models import User, Instance, Key
-from jormungandr import cache, app as current_app
+from jormungandr import cache, app as current_app, app
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/authentication.py]: so, we import `app` two time

File: source/jormungandr/jormungandr/interfaces/v1/test/elastic_search_tests.py [hunk 4]
```diff
@@ -636,9 +636,9 @@ def bragi_make_params_without_instance_test():
         "count": 20
     }
 
-    params = bragi.make_params(request=request, instance=None)
+    params = bragi.make_params(request=request, instances=[])
     rsp = {'q': 'aa', 'limit': 20}
-    len(list(rsp.keys())) == len(list(params.keys()))
+    assert len(list(rsp.keys())) == len(list(params.keys()))
```
Comment (xlqian) [inline source/jormungandr/jormungandr/interfaces/v1/test/elastic_search_tests.py:641]: 💯

**Review by pbougue — APPROVED** [review]

**Pull request merged**